clients/drcachesim/tests/offline-burst_aarch64_sys.templatex

```diff
@@ -13,7 +13,7 @@ Total counts:
      .* total data loads
```

Comment:
I looked back at the details and I really do not understand what is happening with this.  Could you explain how this scenario occurred?  I wrote some of my confusion at https://github.com/DynamoRIO/dynamorio/issues/7050#issuecomment-2443133453.

Putting aside the fact that a regular asynch signal should not cause this kind of thing: if there were some real preempt from say thread relocation, why isn't raw2trace filling in the rest of the instructions in the block?  How is the handler code running already?  That makes it sound like raw2trace is already truncating the rest of the block and somehow solving #5790?  On that note is this PR as written solving #5790?

Reply:
What is in the raw trace records?  What is the instruction count for the `shl` block?  What is the exact raw trace order of the branch's block PC, shl block PC, shl address, and signal marker?

Reply:
Here's the trace without the change:
```
        1436         982:       94080 ifetch       4 byte(s) @ 0x0000aaaae1534568 54000061   b.ne   $0x0000aaaae1534574 (untaken)
        1437         983:       94080 ifetch       4 byte(s) @ 0x0000aaaae153456c 97ffffbd   bl     $0x0000aaaae1534460 -> %x30
        1438         984:       94080 ifetch       4 byte(s) @ 0x0000aaaae1534460 a9be7bfd   stp    %x29 %x30 %sp $0xffffffffffffffe0 -> -0x20(%sp)[16byte] %sp
        1439         984:       94080 write       16 byte(s) @ 0x0000fffffb68e260 by PC 0x0000aaaae1534460
        1440         985:       94080 ifetch       4 byte(s) @ 0x0000aaaae1534464 910003fd   add    %sp $0x0000 lsl $0x00 -> %x29
        1441         986:       94080 ifetch       4 byte(s) @ 0x0000aaaae1534468 90001bc0   adrp   <rel> 0x0000aaaae18ac000 -> %x0
        1442         987:       94080 ifetch       4 byte(s) @ 0x0000aaaae153446c f944dc00   ldr    +0x09b8(%x0)[8byte] -> %x0
        1443         987:       94080 read         8 byte(s) @ 0x0000aaaae18ac9b8 by PC 0x0000aaaae153446c
        1444         988:       94080 ifetch       4 byte(s) @ 0x0000aaaae1534470 f9400001   ldr    (%x0)[8byte] -> %x1
        1445         988:       94080 read         8 byte(s) @ 0x0000fffeddbabb58 by PC 0x0000aaaae1534470
        1446         989:       94080 ifetch       4 byte(s) @ 0x0000aaaae1534474 f9000fe1   str    %x1 -> +0x18(%sp)[8byte]
        1447         989:       94080 write        8 byte(s) @ 0x0000fffffb68e278 by PC 0x0000aaaae1534474
        1448         990:       94080 ifetch       4 byte(s) @ 0x0000aaaae1534478 d2800001   movz   $0x0000 lsl $0x00 -> %x1
        1449         991:       94080 ifetch       4 byte(s) @ 0x0000aaaae153447c b90017ff   str    %wzr -> +0x14(%sp)[4byte]
        1450         991:       94080 write        4 byte(s) @ 0x0000fffffb68e274 by PC 0x0000aaaae153447c
        1451         992:       94080 ifetch       4 byte(s) @ 0x0000aaaae1534480 910053e0   add    %sp $0x0014 lsl $0x00 -> %x0
        1452         993:       94080 ifetch       4 byte(s) @ 0x0000aaaae1534484 d5087620   dc_ivac (%x0)[1byte]
        1453         993:       94080 dflush      64 byte(s) @ 0x0000fffffb68e274 by PC 0x0000aaaad5087620
        1454         993:       94080 <marker: kernel xfer from 0xaaaae1534484 to handler>
        1455         993:       94080 <marker: signal #4>
        1456         993:       94080 <marker: timestamp 13374886167543675>
        1457         993:       94080 <marker: tid 94080 on core 53>
        1458         994:       94080 ifetch       4 byte(s) @ 0x0000aaaae1534384 a9be7bfd   stp    %x29 %x30 %sp $0xffffffffffffffe0 -> -0x20(%sp)[16byte] %sp
        1459         994:       94080 write       16 byte(s) @ 0x0000fffffb68cff0 by PC 0x0000aaaae1534384
```

Raw trace:

```
0010c0 0000fffffb68e29c 0000fffffb68e29c
0010d0 200200000006456c 2024000000064460 1437         983:       94080 ifetch       4 byte(s) @ 0x0000aaaae153456c 97ffffbd bl     $0x0000aaaae1534460 -> %x30
                                         1438         984:       94080 ifetch       4 byte(s) @ 0x0000aaaae1534460 a9be7bfd stp    %x29 %x30 %sp $0xffffffffffffffe0 -> -0x20(%sp)[16byte] %sp
0010e0 0000fffffb68e260 0000aaaae18ac000
0010f0 0000fffeddbabb58 0000fffffb68e278
001100 0000fffffb68e274 0000fffffb68e274 1450         991:       94080 write        4 byte(s) @ 0x0000fffffb68e274 by PC 0x0000aaaae153447c
                                         1450         991:       94080 write        4 byte(s) @ 0x0000fffffb68e274 by PC 0x0000aaaae153447c
001110 c200aaaae1534484 c228000000000004 1454         993:       94080 <marker: kernel xfer from 0xaaaae1534484 to handler>
                                         1455         993:       94080 <marker: signal #4>
001120 802f84635aed7f7b c203000000000035
001130 2022000000064384 0000fffffb68cff0
001140 0000fffffb68d00c 0000aaaae18bde60
001150 0000aaaae18bde60 0000aaaae18bde60
```

Reply:
The plan is:

- File a separate bug on understanding why an asynchronous signal is delivered mid-block; that won't block changing the raw2trace behavior
- We decided we do want to roll back on asynchronous signals
- The only question is for synchronous signals do we want to roll back and also insert some kind of marker: since there are cases like INT3 where the continuation PC is the next PC so you'd never see the INT3 PC.

Reply:
I feel like we have not 100% figured out what we want to do yet, so I don't feel informed enough to review this yet.

As noted above, what is the plan for synchronous signals?

Also, do we want to first understand why the asynch are delivered mid-block: if that turns out to be some problematic issue in DR and we change it, would that affect the decisions here?

Reply:
Let me investigate why the async signals are delivered mid-block,